File: core/ucp/apps/ucp-api/src/ports/outbound/webhook.repository.ts
import { Webhook } from '../../domain/models/webhook.model.js';

export const WEBHOOK_REPOSITORY = 'WEBHOOK_REPOSITORY';

export interface IWebhookRepository {
  save(webhook: Webhook, idempotencyKey?: string): Promise<void>;
  findById(tenantId: string, id: string): Promise<Webhook | null>;
  findAllByTenant(tenantId: string): Promise<Webhook[]>;
  delete(tenantId: string, id: string, idempotencyKey?: string): Promise<void>;
}

File: apps/edi/packages/edi/src/edi/ports/outbound/webhook_repository.py
from collections.abc import Sequence
from typing import Protocol

from edi.domain.models.webhooks import WebhookDomainModel


class WebhookRepositoryPort(Protocol):
    """
    Port for accessing Webhook configurations.
    """

    async def get_webhook(self, tenant_id: str, webhook_id: str) -> WebhookDomainModel | None:
        """
        Fetches Webhook partner config and returns a typed DTO.
        """
        ...

    async def list_webhooks(self, tenant_id: str) -> Sequence[WebhookDomainModel]: ...
    async def save(self, aggregate: WebhookDomainModel) -> None: ...
    async def delete(self, aggregate: WebhookDomainModel) -> None: ...
